service/grpc/codelingo/codelingo.pb.go

```diff
@@ -258,13 +243,188 @@ func (m *GenFact) GetFactName() string {
+}
+
+func (m *GenProperty) Reset()                    { *m = GenProperty{} }
+func (m *GenProperty) String() string            { return proto.CompactTextString(m) }
+func (*GenProperty) ProtoMessage()               {}
+func (*GenProperty) Descriptor() ([]byte, []int) { return fileDescriptor0, []int{6} }
+
+type isGenProperty_Value interface {
+	isGenProperty_Value()
+}
+
+type GenProperty_String_ struct {
+	String_ string `protobuf:"bytes,1,opt,name=string,oneof"`
+}
+type GenProperty_Int struct {
+	Int int64 `protobuf:"varint,2,opt,name=int,oneof"`
+}
+type GenProperty_Float struct {
+	Float float32 `protobuf:"fixed32,3,opt,name=float,oneof"`
+}
+type GenProperty_Bool struct {
+	Bool bool `protobuf:"varint,4,opt,name=bool,oneof"`
+}
+
+func (*GenProperty_String_) isGenProperty_Value() {}
+func (*GenProperty_Int) isGenProperty_Value()     {}
+func (*GenProperty_Float) isGenProperty_Value()   {}
+func (*GenProperty_Bool) isGenProperty_Value()    {}
+
+func (m *GenProperty) GetValue() isGenProperty_Value {
+	if m != nil {
+		return m.Value
+	}
+	return nil
+}
+
+func (m *GenProperty) GetString_() string {
+	if x, ok := m.GetValue().(*GenProperty_String_); ok {
+		return x.String_
+	}
+	return ""
+}
+
+func (m *GenProperty) GetInt() int64 {
+	if x, ok := m.GetValue().(*GenProperty_Int); ok {
+		return x.Int
+	}
+	return 0
+}
+
+func (m *GenProperty) GetFloat() float32 {
+	if x, ok := m.GetValue().(*GenProperty_Float); ok {
+		return x.Float
+	}
+	return 0
+}
+
+func (m *GenProperty) GetBool() bool {
+	if x, ok := m.GetValue().(*GenProperty_Bool); ok {
+		return x.Bool
+	}
+	return false
+}
+
+// XXX_OneofFuncs is for the internal use of the proto package.
+func (*GenProperty) XXX_OneofFuncs() (func(msg proto.Message, b *proto.Buffer) error, func(msg proto.Message, tag, wire int, b *proto.Buffer) (bool, error), func(msg proto.Message) (n int), []interface{}) {
+	return _GenProperty_OneofMarshaler, _GenProperty_OneofUnmarshaler, _GenProperty_OneofSizer, []interface{}{
+		(*GenProperty_String_)(nil),
+		(*GenProperty_Int)(nil),
+		(*GenProperty_Float)(nil),
+		(*GenProperty_Bool)(nil),
+	}
+}
+
+func _GenProperty_OneofMarshaler(msg proto.Message, b *proto.Buffer) error {
+	m := msg.(*GenProperty)
+	// value
+	switch x := m.Value.(type) {
+	case *GenProperty_String_:
+		b.EncodeVarint(1<<3 | proto.WireBytes)
+		b.EncodeStringBytes(x.String_)
+	case *GenProperty_Int:
+		b.EncodeVarint(2<<3 | proto.WireVarint)
+		b.EncodeVarint(uint64(x.Int))
+	case *GenProperty_Float:
+		b.EncodeVarint(3<<3 | proto.WireFixed32)
+		b.EncodeFixed32(uint64(math.Float32bits(x.Float)))
+	case *GenProperty_Bool:
+		t := uint64(0)
+		if x.Bool {
+			t = 1
+		}
+		b.EncodeVarint(4<<3 | proto.WireVarint)
+		b.EncodeVarint(t)
+	case nil:
+	default:
+		return fmt.Errorf("GenProperty.Value has unexpected type %T", x)
+	}
+	return nil
+}
+
+func _GenProperty_OneofUnmarshaler(msg proto.Message, tag, wire int, b *proto.Buffer) (bool, error) {
```

Comment:
don't use underscores in Go names; func _GenProperty_OneofUnmarshaler should be _GenPropertyOneofUnmarshaler